tools/package-source.js

```diff
@@ -99,6 +100,75 @@ var splitConstraint = function (c) {
            constraint: parsed.constraintString || null };
 };
 
+// Given the text of a README.md file, excerpts the text between the first and
+// second heading.
+//
+// Specifically - if there is text between the document name, and the first
+// subheading, it will take that text. If there is no text there, and only text
+// after the first subheading, it will take that text. It won't look any deeper
+// than that (in case the user intentionally wants to leave the section blank
+// for some reason). Skips lines that start with an exclamation point.
+var getExcerptFromReadme = function (text) {
+  // Don't waste time parsing if the document is empty.
+  if (! text) return "";
+
+  // Split into linesarse with Commonmark.
+  var reader = new commonmark.DocParser();
+  var parsed = reader.parse(text);
+
+  // Commonmark will parse the Markdown into an array of nodes. These are the
+  // nodes that represent the text between the first and second heading.
+  var relevantNodes = [];
+
+  // We want to take the data between the first and second headers, so it is
+  // important to keep track of how many headers we have passed.
+  var headerNum = 0;
+
+  // Go through the document until we get the nodes that we are looking for,
+  // then stop.
+  _.any(parsed.children, function (child) {
+    var isHeader = (child.t === "Header");
+    // Don't excerpt anything before the first header.
+    if (! isHeader && (headerNum > 0)) {
+      // If we are currently in the middle of excerpting, continue doing that
+      // until we hit hit a header (and this is not a header). Otherwise, if
+      // this is text, we should begin to excerpt it.
+      relevantNodes.push(child);
+    } else if (! _.isEmpty(relevantNodes) && isHeader) {
+      // We have been excerpting, and came across a header. That means
+      // that we are done.
+      return true;
+    } else if (isHeader) {
+      // We are going to take the first text that we encounter under either of
```

Comment:
That's exactly what I was going for, but, honestly, I think you are right; I am overcomplicating it and I am going to change it.